node/src/tasks/user_sends_file.rs

```diff
@@ -48,6 +51,81 @@ where
             storage_hub_handler,
         }
     }
+
+    async fn send_chunks_to_provider(
```

Comment:
I'd prefer to have these utility functions at the bottom of the file, so that it is easier to see what events this task handles, and therefore how is the flow.

Reply:
That means I would have to declare another impl block for `UserSendsFileTask`. Or would you like me to move the current impl to the bottom of the file?